src/entities/nestedJoin/getNestedJoinCalls.ts

```diff
@@ -0,0 +1,70 @@
+import { Token } from '../token';
+import { getPoolAddress } from '../../utils';
+import { NestedJoinInput, NestedJoinCall } from './types';
+import { NestedPoolState } from '../types';
+
+export const getNestedJoinCalls = (
+    {
+        amountsIn,
+        chainId,
+        accountAddress,
+        useNativeAssetAsWrappedAmountIn,
+        fromInternalBalance,
+    }: NestedJoinInput,
+    { pools }: NestedPoolState,
+): NestedJoinCall[] => {
+    /**
+     * Overall logic to build sequence of join calls:
+     * 1. Go from bottom pool to up filling out input amounts and output refs
+     * 2. Inputs will be amountsIn provided, output of the previous level or 0n
+     * 3. Output at max level is the bptOut
+     */
+
+    const poolsSortedByLevel = pools.sort((a, b) => a.level - b.level);
+
+    const calls: NestedJoinCall[] = [];
+    for (const pool of poolsSortedByLevel) {
+        const sortedTokens = pool.tokens
+            .sort((a, b) => a.index - b.index)
+            .map((t) => new Token(chainId, t.address, t.decimals));
+        calls.push({
+            chainId: chainId,
+            useNativeAssetAsWrappedAmountIn:
+                useNativeAssetAsWrappedAmountIn ?? false,
+            sortedTokens,
+            poolId: pool.id,
+            poolType: pool.type,
+            kind: 0,
+            sender: accountAddress,
```

Comment:
I think the multicall could be more efficient by using relayer and internal balances right? (Can be done in a future PR but just wanted to clarify at this stage)

Reply:
We should be able to use the relayer to reduce gas cost 👍 
But chaining actions with internal balances require each action to have both `from` and `to` internal balance available, which is not the case for `joins` and `exits`. The previous implementation used it mainly with `swaps`.
The best we can do is to have the first join action with `fromInternalBalance = true` and the last exit action with `toInternalBalance = true` - I'll go ahead and make that change.

Reply:
Regarding `from/to internal balance`, on the previous implementation, we only used for intermediary calls. A join with fromInternalBalance where the user's balance was originally internal was not supported.
Is this something we should support on b-sdk?
Being more concrete:
- example of flow we supported: non-internal -> internal -> internal -> non-internal
- example of flow we didn't support: internal -> internal -> non-internal
I ask because it adds a bit of complexity on tests. Nothing too complex, but worth checking before implementing.

Reply:
We definitely don't need to support that new user case.